dev-infra/pullapprove/group.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {IMinimatch, Minimatch, match} from 'minimatch';
+
+import {PullApproveGroupConfig} from './parse-yaml';
+
+/** A condition for a group. */
+interface GroupCondition {
+  glob: string;
+  matcher: IMinimatch;
+  matchedFiles: Set<string>;
+}
+
+/** Result of testing files against the group. */
+export interface PullApproveGroupResult {
+  groupName: string;
+  matchedIncludes: GroupCondition[];
+  matchedExcludes: GroupCondition[];
+  matchedCount: number;
+  unmatchedIncludes: GroupCondition[];
+  unmatchedExcludes: GroupCondition[];
+  unmatchedCount: number;
+}
+
+// Regex Matcher for contains_any_globs conditions
+const CONTAINS_ANY_GLOBS_REGEX = /^'([^']+)',?$/;
+
+/** A PullApprove group to be able to test files against. */
+export class PullApproveGroup {
+  // Lines which were not able to be parsed as expected.
+  private misconfiguredLines: string[] = [];
+  // Conditions for the group for including files.
+  private includeConditions: GroupCondition[] = [];
+  // Conditions for the group for excluding files.
+  private excludeConditions: GroupCondition[] = [];
+  // Whether the group has file matchers.
+  public hasMatchers = false;
+
+  constructor(public groupName: string, group: PullApproveGroupConfig) {
+    for (let condition of group.conditions) {
+      condition = condition.trim();
+
+      if (condition.startsWith('contains_any_globs')) {
+        const [conditions, misconfiguredLines] = getLinesForContainsAnyGlobs(condition);
+        conditions.forEach(globString => this.includeConditions.push({
+          glob: globString,
+          matcher: new Minimatch(globString, {dot: true}),
+          matchedFiles: new Set<string>(),
+        }));
+        this.misconfiguredLines.push(...misconfiguredLines);
+        this.hasMatchers = true;
+        continue;
+      }
+
+      if (condition.startsWith('not contains_any_globs')) {
```

Comment:
Nit: The `continue` statements are redundant here (and below).

```suggestion
      } else if (condition.startsWith('not contains_any_globs')) {
```